Document how to see logger output when running the twisted tests (#10148)
matrix-org/synapse: CONTRIBUTING.md

```sh
source ./env/bin/activate
trial tests
# The tests can also be run with this longer command

> python -m twisted.trial tests

Author: I think it's good to call this out because the readme uses this syntax and people might be Ctrl+F finding where to jump off from.

Reviewer: I see your point, though I'm bound to say that I think this just makes it confusing for readers of CONTRIBUTING.md.

A better solution would be to use one command or the other consistently. (I'd suggest `trial` since it's shorter, but ymmv)

Author: I've updated the readme to use `trial tests` 🤷

trial tests.rest.admin.test_room tests.handlers.test_admin.ExfiltrateData.test_invite
```

If your tests fail, you may wish to look at the logs (the default log level is `ERROR`):

```sh

> less _trial_temp/test.log

Author: It would be interesting to add a note to the `trial tests` output when the tests fail to go checkout the `_trial_temp/test.log` file. I wonder if we can hook in at all 🤔

Author: It looks possible if we defined our own reporter but then that requires adding `--reporter=ourcustomreporter` to the test command.

Anyway, another possible PR ⏩

less _trial_temp/test.log
```

> To see always see logger output from the app while the tests run,
> set `SYNAPSE_TEST_LOG_LEVEL` and the resulting logs will be in `_trial_temp/test.log`:

Reviewer: This sentence doesn't make any sense. (Also, we've just mentioned `_trial_temp` - do we really need it again? Leave it in if you prefer, though).

```suggestion
To increase the log level for the tests,  set `SYNAPSE_TEST_LOG_LEVEL`:
```
